src/components/shared/TreeView.js

```diff
@@ -512,48 +518,69 @@ class TreeView<
       return;
     }
 
-    if (isArrowKey) {
-      switch (event.keyCode) {
-        case 37: // KEY_LEFT
-          {
-            const isCollapsed = this._isCollapsed(selected);
-            if (!isCollapsed) {
-              this._toggle(selected);
-            } else {
-              const parent = this.props.tree.getParent(selected);
-              if (parent !== -1) {
-                this._select(parent);
-              }
+    if (isNavigationKey) {
+      switch (event.key) {
+        case 'ArrowLeft': {
+          const isCollapsed = this._isCollapsed(selected);
+          if (!isCollapsed) {
+            this._toggle(selected);
+          } else {
+            const parent = this.props.tree.getParent(selected);
+            if (parent !== -1) {
+              this._select(parent);
             }
           }
           break;
-        case 38: // KEY_UP
-          {
-            if (selectedRowIndex > 0) {
-              this._select(visibleRows[selectedRowIndex - 1]);
-            }
+        }
+        case 'ArrowUp': {
+          if (selectedRowIndex > 0) {
+            this._select(visibleRows[selectedRowIndex - 1]);
           }
           break;
-        case 39: // KEY_RIGHT
-          {
-            const isCollapsed = this._isCollapsed(selected);
-            if (isCollapsed) {
-              this._toggle(selected);
-            } else {
-              // Do KEY_DOWN only if the next element is a child
-              if (this.props.tree.hasChildren(selected)) {
-                this._select(this.props.tree.getChildren(selected)[0]);
-              }
-            }
+        }
+        case 'PageUp': {
```

Comment:
This doesn't work on my MacBook. `Cmd + Up` is the correct shortcut.

Page up: `event.metaKey && event.key === "ArrowUp"`
Page down: `event.metaKey && event.key === "ArrowDown"`

Reply:
I'm so sad.

Reply:
Yeah I don't think `Cmd + Up` is the right shortcut. It doesn't do page scrolling in Activity Monitor or in Thunderbird, for example. Here's the key handling code of the XUL tree view, which we could use for inspiration: https://searchfox.org/mozilla-central/rev/8384a6519437f5eefbe522196f9ddf5c8b1d3fb4/toolkit/content/widgets/tree.xml#822-905

Reply:
I can implement Cmd + Up/Down for sure, but I'd rather keep `Fn + left/right` as it is already in the patch, because then I don't need to implement different behavior for different OS. (maybe we'll have to eventually).

Also I believe that text editors are different. On my OS, Home and End doesn't move up/down in a text area, rather they go respectively at the start and end of a line. It would be nice to know how these shortcuts work for you on apps and contexts that are more list-based, eg Apple Mail or Firefox' bookmarks list.

Reply:
From https://support.apple.com/en-us/HT201236:
```
Fn–Up Arrow: Page Up: Scroll up one page. 
Fn–Down Arrow: Page Down: Scroll down one page.
Fn–Left Arrow: Home: Scroll to the beginning of a document.
Fn–Right Arrow: End: Scroll to the end of a document.
Command–Up Arrow: Move the insertion point to the beginning of the document.
Command–Down Arrow: Move the insertion point to the end of the document.
Command–Left Arrow: Move the insertion point to the beginning of the current line.
Command–Right Arrow: Move the insertion point to the end of the current line.
```
(which is what you describe, with more informaiton about other shortcuts)

Reply:
I think eventually I'd like to make Home work like this:
1. if the current selected node isn't a root node, go to the root node of the current selected node.
2. if the current selected node is a root node, go to the top of the tree.
Because this is a bit more work I left this out of this PR initially, but please tell me what you think :)

Reply:
I'm not sure how End would work in this context though. Maybe the behavior outlined in #517 ?